tests/unit/components/gallery/HDGallery.spec.js

```diff
@@ -1,33 +1,111 @@
 import { wrapperFactoryBuilder } from 'tests/unit/helpers';
 import HdGallery from '@/components/gallery/HdGallery.vue';
+import HdGalleryPlaceholder from '@/components/gallery/HdGalleryPlaceholder.vue';
 import ITEMS from '@/stories/mocks/GALLERY_ITEMS';
 
-const IMAGE_SELECTOR = '.gallery__image';
-
-const wrapperBuilder = wrapperFactoryBuilder(HdGallery, {
-  props: {
-    items: ITEMS,
-  },
-});
-
 describe('HdGallery', () => {
-  it('renders component', () => {
-    const wrapper = wrapperBuilder();
+  const build = (props) => {
+    const wrapper = wrapperFactoryBuilder(HdGallery, {
+      props: {
+        items: ITEMS,
+      },
+    })({ props });
 
+    return {
+      wrapper,
+      caption: () => wrapper.find('.gallery__caption'),
+      nextButton: () => wrapper.find('.gallery__controls-next'),
+      previousButton: () => wrapper.find('.gallery__controls-prev'),
+      selectedPicture: () => wrapper.find('.gallery-media__object-picture'),
+      selectedPictureImg: () => wrapper.find('.gallery-media__object-picture').find('img'),
+      selectedPictureInfo: () => wrapper.find('.gallery__info'),
+      carousel: () => wrapper.findAll('.gallery__carousel'),
+      carouselWrap: () => wrapper.findAll('.gallery-carousel__wrap'),
+      flickity: () => wrapper.find('.gallery-carousel').find({ ref: 'flickity' }),
+      placeholder: () => wrapper.find(HdGalleryPlaceholder),
```

Comment:
I thought about those being something from the factory as well. Not sure how it would work but something like:

```javascript
const wrapperFactory = wrapperFactoryBuilder(Component, { props, data... }, { getterFunction1, getterFunction2...});
```

or

```javascript
const wrapperFactory = wrapperFactoryBuilder(Component, { props, data, wrapperGetterFunctions... });
```

Reply:
This is a nice idea, if you want we can discuss it a bit more.

My only worry is, by passing the selectors as parameters to the builder, we are supposed to change the value returned by the factory, and this might change all tests.

What do you think?

Reply:
Hummm... It should be an enhancement only otherwise we will be painful hahah. Maybe it could be injected in the `wrapper`?

Nevertheless, please, do not do it in this PR. We can bring this topic to team sync. It has to be an approach that must of us agree at least.

Reply:
I will write it down and we can bring that in the sync then =D